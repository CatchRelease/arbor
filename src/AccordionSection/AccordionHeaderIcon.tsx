import { FC } from 'react';

import Icon, { IconProps } from '../Icon';

type Props = Omit<IconProps, 'name'> & {
  isOpen: boolean | undefined;
};

const AccordionHeaderIcon: FC<Props> = ({ isOpen, ...props }) => (
  <Icon mr="small" name="chevron" rotation={isOpen ? null : '270'} {...props} />
);

export default AccordionHeaderIcon;
