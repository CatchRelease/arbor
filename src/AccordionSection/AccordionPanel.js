import PropTypes from 'prop-types';

import StyledAccordionPanel from './StyledAccordionPanel';

const AccordionPanel = ({ isOpen, ...panelProps }) => {
  const display = isOpen ? 'block' : 'none';

  return <StyledAccordionPanel {...{ ...panelProps, display }} />;
};

AccordionPanel.propTypes = {
  /**
   * Whether or not the panel is open or not. When the panel is closed, the
   * content will be rendered with display none.
   */
  isOpen: PropTypes.bool.isRequired
};

export default AccordionPanel;
