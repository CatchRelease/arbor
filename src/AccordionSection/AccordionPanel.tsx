import { FC } from 'react';

import StyledAccordionPanel, {
  Props as StyledAccordionPanelProps
} from './StyledAccordionPanel';

type Props = StyledAccordionPanelProps & {
  isOpen: boolean | undefined;
};

const AccordionPanel: FC<Props> = ({ isOpen, ...panelProps }) => {
  const display = isOpen ? 'block' : 'none';

  return <StyledAccordionPanel {...{ ...panelProps, display }} />;
};

export default AccordionPanel;
