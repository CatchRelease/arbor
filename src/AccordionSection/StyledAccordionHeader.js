import styled from '@emotion/styled';

import createWithComponent from '../utils/createWithComponent';
import Flex from '../Flex';

const StyledHeader = styled(createWithComponent(Flex, 'header'))`
  background: ${props => props.theme.colors.background.default};
  border-bottom: 1px solid ${props => props.theme.borderColors.default};

  section:last-child & {
    border-bottom: none;
  }

  .button--minimal:last-of-type {
    margin-right: -${props => props.theme.space.small};

    + :not(object) {
      margin-left: ${props => props.theme.space.small};
    }
  }

  button + .button--secondary,
  button + .button--primary {
    margin-left: ${props => props.theme.space.smaller};
  }
`;

export default StyledHeader;
