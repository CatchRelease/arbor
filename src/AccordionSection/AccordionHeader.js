import PropTypes from 'prop-types';

import StyledAccordionHeader from './StyledAccordionHeader';
import StyledAccordionHeading from './StyledAccordionHeading';
import StyledAccordionHeadingText from './StyledAccordionHeadingText';
import Icon from '../Icon';
import VARIANTS from './variants';

const AccordionHeaderIcon = ({ isOpen, ...props }) => (
  <Icon mr="small" name="chevron" rotation={isOpen ? null : '270'} {...props} />
);

AccordionHeaderIcon.propTypes = {
  isOpen: PropTypes.bool.isRequired
};

const AccordionHeader = ({
  isOpen,
  note,
  onClick,
  panelId,
  text,
  iconRight,
  ...props
}) => (
  <StyledAccordionHeader
    alignItems="center"
    justifyContent="space-between"
    px="regular"
    onClick={onClick}
    aria-controls={panelId}
    aria-expanded={isOpen}
    {...props}
  >
    <StyledAccordionHeading fontWeight="medium" mr="regular">
      <StyledAccordionHeadingText
        alignItems="center"
        aria-controls={panelId}
        aria-expanded={isOpen}
      >
        {iconRight ? (
          <>
            {text}
            <AccordionHeaderIcon isOpen={isOpen} ml="smallest" />
          </>
        ) : (
          <>
            <AccordionHeaderIcon isOpen={isOpen} />
            {text}
          </>
        )}
      </StyledAccordionHeadingText>
    </StyledAccordionHeading>
    {note}
  </StyledAccordionHeader>
);

AccordionHeader.propTypes = {
  /**
   * Whether or not the AccordionSection this header resides in is open or not.
   */
  isOpen: PropTypes.bool.isRequired,

  /**
   * Optional note to render within the AccordionHeader
   */
  note: PropTypes.node,

  /**
   * Callback to call whenever the StyledAccordionHeadingButton is clicked
   */
  onClick: PropTypes.func,

  /**
   * ID of the AccordionPanel that holds the content of the AccordionSection.
   * Used to specify aria-controls for accessbility.
   */
  panelId: PropTypes.string.isRequired,

  /**
   * Header text for the section
   */
  text: PropTypes.node.isRequired,

  /**
   * Link variant for styling. See storybook for examples.
   */
  variant: PropTypes.oneOf(VARIANTS).isRequired,

  /**
   * Optional flag to position icon to right of header text
   */
  iconRight: PropTypes.bool
};

AccordionHeader.defaultProps = {
  note: null,
  onClick: () => {},
  iconRight: false
};

export default AccordionHeader;
