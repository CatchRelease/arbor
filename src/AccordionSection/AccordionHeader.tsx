import { FC, MouseEventHandler, ReactNode } from 'react';

import StyledAccordionHeader from './StyledAccordionHeader';
import StyledAccordionHeading from './StyledAccordionHeading';
import StyledAccordionHeadingText from './StyledAccordionHeadingText';

import AccordionHeaderIcon from './AccordionHeaderIcon';
import { Variant } from './variants';

type Props = {
  iconRight?: boolean;
  isOpen: boolean | undefined;
  note?: ReactNode;
  onClick?: MouseEventHandler<HTMLElement>;
  panelId: string;
  text: ReactNode;
  variant?: Variant;
};

const AccordionHeader: FC<Props> = ({
  isOpen,
  note,
  onClick = () => {},
  panelId,
  text,
  iconRight = false,
  ...props
}) => (
  <StyledAccordionHeader
    alignItems="center"
    justifyContent="space-between"
    px="regular"
    onClick={onClick}
    aria-controls={panelId}
    aria-expanded={isOpen}
    {...props}
  >
    <StyledAccordionHeading fontSize="size4" fontWeight="medium" mr="regular">
      <StyledAccordionHeadingText
        alignItems="center"
        aria-controls={panelId}
        aria-expanded={isOpen}
      >
        {iconRight ? (
          <>
            {text}
            <AccordionHeaderIcon isOpen={isOpen} ml="smallest" />
          </>
        ) : (
          <>
            <AccordionHeaderIcon isOpen={isOpen} />
            {text}
          </>
        )}
      </StyledAccordionHeadingText>
    </StyledAccordionHeading>
    {note}
  </StyledAccordionHeader>
);

export default AccordionHeader;
