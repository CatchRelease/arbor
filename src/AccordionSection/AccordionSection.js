import React from 'react';
import PropTypes from 'prop-types';

import AccordionHeader from './AccordionHeader';
import AccordionPanel from './AccordionPanel';

class AccordionSection extends React.Component {
  state = {
    isOpen: false
  };

  get isControlled() {
    const { isOpen } = this.props;

    return isOpen !== undefined;
  }

  get isOpen() {
    const { isOpen: controlledIsOpen } = this.props;
    const { isOpen: internalIsOpen } = this.state;

    return this.isControlled ? controlledIsOpen : internalIsOpen;
  }

  onHeaderClick = () => {
    const { onHeaderClick } = this.props;

    if (!this.isControlled) {
      this.toggle();
    }

    onHeaderClick();
  };

  toggle() {
    this.setState(prevState => ({ isOpen: !prevState.isOpen }));
  }

  render() {
    const { children, header, headerNote, panelId } = this.props;

    return (
      <section>
        <AccordionHeader
          isOpen={this.isOpen}
          note={headerNote}
          onClick={this.onHeaderClick}
          panelId={panelId}
          text={header}
        />
        <AccordionPanel id={panelId} isOpen={this.isOpen}>
          {children}
        </AccordionPanel>
      </section>
    );
  }
}

AccordionSection.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node)
  ]).isRequired,
  header: PropTypes.node.isRequired,
  headerNote: PropTypes.node,
  isOpen: PropTypes.bool,
  onHeaderClick: PropTypes.func,
  panelId: PropTypes.string.isRequired
};

AccordionSection.defaultProps = {
  isOpen: undefined,
  headerNote: null,
  onHeaderClick: () => {}
};

export default AccordionSection;
