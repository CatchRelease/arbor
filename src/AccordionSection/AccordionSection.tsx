import { FC, ReactNode, useState } from 'react';

import AccordionHeader from './AccordionHeader';
import AccordionPanel from './AccordionPanel';
import { Variant } from './variants';

export type Props = {
  header: ReactNode;
  headerNote?: ReactNode;
  isOpen?: boolean;
  onHeaderClick?: () => void;
  panelId: string;
  variant?: Variant;
};

const AccordionSection: FC<Props> = ({
  children,
  header,
  headerNote,
  isOpen: controlledIsOpen,
  onHeaderClick = () => {},
  panelId,
  variant = 'default',
  ...props
}) => {
  const [internalIsOpen, setInternalIsOpen] = useState(false);

  const isControlled = controlledIsOpen !== undefined;
  const isOpen = isControlled ? controlledIsOpen : internalIsOpen;

  const toggle = () => {
    setInternalIsOpen((previous) => !previous);
  };

  const handleHeaderClick = () => {
    if (!isControlled) {
      toggle();
    }

    onHeaderClick();
  };

  return (
    <section>
      <AccordionHeader
        isOpen={isOpen}
        note={headerNote}
        onClick={handleHeaderClick}
        panelId={panelId}
        text={header}
        variant={variant}
        {...props}
      />
      <AccordionPanel id={panelId} isOpen={isOpen} variant={variant}>
        {children}
      </AccordionPanel>
    </section>
  );
};

export default AccordionSection;
