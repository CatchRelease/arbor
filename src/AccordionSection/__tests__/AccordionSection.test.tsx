import { shallow } from 'enzyme';

import createWithTheme from '../../../utils/createWithTheme';
import AccordionSection, { Props } from '../AccordionSection';

describe('<AccordionSection />', () => {
  const baseProps = {
    children: <p>Hello World</p>,
    header: 'My Accordion',
    panelId: 'my-accordion-1'
  };

  const render = (additionalProps?: Partial<Props>) => {
    const props = { ...baseProps, ...additionalProps };
    return shallow(<AccordionSection {...props} />);
  };

  describe('Variants', () => {
    (['default', 'minimal'] as const).forEach((variant) => {
      it(`properly renders a ${variant} accordion`, () => {
        const accordion = createWithTheme(
          <AccordionSection {...{ ...baseProps, variant }} />
        );

        expect(accordion).toMatchSnapshot();
      });
    });
  });

  describe('render', () => {
    const createTree = (additionalProps = {}) =>
      createWithTheme(
        <AccordionSection {...{ ...baseProps, ...additionalProps }} />
      );

    context('accordion is open', () => {
      it('renders with the panel visible', () => {
        const tree = createTree({
          isOpen: true
        });

        expect(tree).toMatchSnapshot();
      });
    });

    context('accordion is closed', () => {
      it('renders with the panel hidden', () => {
        const tree = createTree({
          isOpen: false
        });

        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('Children', () => {
    describe('AccordionHeader', () => {
      it('renders an AccordionHeader', () => {
        const section = render();

        expect(section.find('AccordionHeader')).toExist();
      });

      it('passes a note to the accordion header', () => {
        const headerNote = 'cordions are fun';
        const section = render({ headerNote });

        expect(section.find('AccordionHeader')).toHaveProp({
          note: headerNote
        });
      });

      it('passes a header to the accordion header', () => {
        const header = 'Accordion header';
        const section = render({ header });

        expect(section.find('AccordionHeader')).toHaveProp({ text: header });
      });
    });

    context('Controlled accordion', () => {
      context('accordion is open', () => {
        it('opens the accordion', () => {
          const section = render({ isOpen: true });

          expect(section.find('AccordionPanel')).toHaveProp({ isOpen: true });
        });
      });

      context('accordion is closed', () => {
        it('does not render the accordion body', () => {
          const section = render({ isOpen: false });

          expect(section.find('AccordionPanel')).toHaveProp({ isOpen: false });
        });
      });
    });

    context('Uncontrolled accordion', () => {
      it('does not render the accordion body', () => {
        const section = render();

        expect(section.find('AccordionPanel')).toHaveProp({ isOpen: false });
      });
    });
  });

  describe('event handlers', () => {
    describe('accordion header onClick', () => {
      context('uncontrolled accordion', () => {
        it('opens and closes the accordion', () => {
          const section = render();

          section.find('AccordionHeader').simulate('click');

          expect(section.find('AccordionPanel')).toHaveProp({ isOpen: true });

          section.find('AccordionHeader').simulate('click');

          expect(section.find('AccordionPanel')).toHaveProp({ isOpen: false });
        });
      });

      context('controlled accordion', () => {
        it('does not change the accordion state', () => {
          const section = render({ isOpen: false });

          section.find('AccordionHeader').simulate('click');

          expect(section.find('AccordionPanel')).toHaveProp({ isOpen: false });
        });

        it('calls the onHeaderClick prop', () => {
          const onHeaderClick = jest.fn();

          const section = render({ isOpen: false, onHeaderClick });

          section.find('AccordionHeader').simulate('click');

          expect(onHeaderClick).toHaveBeenCalled();
        });
      });
    });
  });
});
