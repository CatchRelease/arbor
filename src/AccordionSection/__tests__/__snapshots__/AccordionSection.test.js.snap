// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionSection /> Variants properly renders a default accordion 1`] = `
.emotion-7 {
  font-family: Motiva Sans;
  padding-left: 1rem;
  padding-right: 1rem;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  cursor: pointer;
  background: #F2F2F2;
  border-bottom: 1px solid #E0E1E2;
}

.emotion-7 .button--minimal:last-of-type {
  margin-right: -0.25rem;
}

.emotion-7 .button--minimal:last-of-type +:not(object) {
  margin-left: 0.25rem;
}

.emotion-7 button+.button--secondary,
.emotion-7 button+.button--primary {
  margin-left: 0.5rem;
}

section:last-child .emotion-7 {
  border-bottom: none;
}

.emotion-5 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.25;
  margin-right: 1rem;
  line-height: 1.25;
  margin-bottom: calc(0.875rem * 0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-bottom: 0;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  font-size: inherit;
  font-weight: 400;
  padding: 0;
}

.emotion-0 {
  margin-right: 0.75rem;
  -webkit-transform: rotate(270deg);
  -moz-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
}

.emotion-9 {
  font-family: Motiva Sans;
  overflow: hidden;
  box-sizing: border-box;
  display: none;
  border-bottom: 1px solid #E0E1E2;
}

section:last-child .emotion-9 {
  border-bottom: none;
  border-top: 1px solid #E0E1E2;
}

<section>
  <header
    aria-controls="my-accordion-1"
    aria-expanded={false}
    className="emotion-7 emotion-8"
    onClick={[Function]}
  >
    <h1
      className="emotion-4 emotion-5 emotion-6"
      color="text.dark"
      fontSize="size4"
      fontWeight="medium"
    >
      <span
        aria-controls="my-accordion-1"
        aria-expanded={false}
        className="emotion-2 emotion-3"
      >
        <i
          className="ar ar-chevron emotion-0 emotion-1"
        />
        My Accordion
      </span>
    </h1>
  </header>
  <div
    className="emotion-9 emotion-10"
    display="none"
    id="my-accordion-1"
    overflow="hidden"
  >
    <p>
      Hello World
    </p>
  </div>
</section>
`;

exports[`<AccordionSection /> Variants properly renders a minimal accordion 1`] = `
.emotion-5 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.25;
  margin-right: 1rem;
  line-height: 1.25;
  margin-bottom: calc(0.875rem * 0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-bottom: 0;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  font-size: inherit;
  font-weight: 400;
  padding: 0;
}

.emotion-0 {
  margin-right: 0.75rem;
  -webkit-transform: rotate(270deg);
  -moz-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
}

.emotion-7 {
  font-family: Motiva Sans;
  padding-left: 1rem;
  padding-right: 1rem;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  cursor: pointer;
}

.emotion-7 .button--minimal:last-of-type {
  margin-right: -0.25rem;
}

.emotion-7 .button--minimal:last-of-type +:not(object) {
  margin-left: 0.25rem;
}

.emotion-7 button+.button--secondary,
.emotion-7 button+.button--primary {
  margin-left: 0.5rem;
}

.emotion-9 {
  font-family: Motiva Sans;
  overflow: hidden;
  box-sizing: border-box;
  display: none;
}

<section>
  <header
    aria-controls="my-accordion-1"
    aria-expanded={false}
    className="emotion-7 emotion-8"
    onClick={[Function]}
  >
    <h1
      className="emotion-4 emotion-5 emotion-6"
      color="text.dark"
      fontSize="size4"
      fontWeight="medium"
    >
      <span
        aria-controls="my-accordion-1"
        aria-expanded={false}
        className="emotion-2 emotion-3"
      >
        <i
          className="ar ar-chevron emotion-0 emotion-1"
        />
        My Accordion
      </span>
    </h1>
  </header>
  <div
    className="emotion-9 emotion-10"
    display="none"
    id="my-accordion-1"
    overflow="hidden"
  >
    <p>
      Hello World
    </p>
  </div>
</section>
`;

exports[`<AccordionSection /> render accordion is closed renders with the panel hidden 1`] = `
.emotion-7 {
  font-family: Motiva Sans;
  padding-left: 1rem;
  padding-right: 1rem;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  cursor: pointer;
  background: #F2F2F2;
  border-bottom: 1px solid #E0E1E2;
}

.emotion-7 .button--minimal:last-of-type {
  margin-right: -0.25rem;
}

.emotion-7 .button--minimal:last-of-type +:not(object) {
  margin-left: 0.25rem;
}

.emotion-7 button+.button--secondary,
.emotion-7 button+.button--primary {
  margin-left: 0.5rem;
}

section:last-child .emotion-7 {
  border-bottom: none;
}

.emotion-5 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.25;
  margin-right: 1rem;
  line-height: 1.25;
  margin-bottom: calc(0.875rem * 0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-bottom: 0;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  font-size: inherit;
  font-weight: 400;
  padding: 0;
}

.emotion-0 {
  margin-right: 0.75rem;
  -webkit-transform: rotate(270deg);
  -moz-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
}

.emotion-9 {
  font-family: Motiva Sans;
  overflow: hidden;
  box-sizing: border-box;
  display: none;
  border-bottom: 1px solid #E0E1E2;
}

section:last-child .emotion-9 {
  border-bottom: none;
  border-top: 1px solid #E0E1E2;
}

<section>
  <header
    aria-controls="my-accordion-1"
    aria-expanded={false}
    className="emotion-7 emotion-8"
    onClick={[Function]}
  >
    <h1
      className="emotion-4 emotion-5 emotion-6"
      color="text.dark"
      fontSize="size4"
      fontWeight="medium"
    >
      <span
        aria-controls="my-accordion-1"
        aria-expanded={false}
        className="emotion-2 emotion-3"
      >
        <i
          className="ar ar-chevron emotion-0 emotion-1"
        />
        My Accordion
      </span>
    </h1>
  </header>
  <div
    className="emotion-9 emotion-10"
    display="none"
    id="my-accordion-1"
    overflow="hidden"
  >
    <p>
      Hello World
    </p>
  </div>
</section>
`;

exports[`<AccordionSection /> render accordion is open renders with the panel visible 1`] = `
.emotion-7 {
  font-family: Motiva Sans;
  padding-left: 1rem;
  padding-right: 1rem;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  cursor: pointer;
  background: #F2F2F2;
  border-bottom: 1px solid #E0E1E2;
}

.emotion-7 .button--minimal:last-of-type {
  margin-right: -0.25rem;
}

.emotion-7 .button--minimal:last-of-type +:not(object) {
  margin-left: 0.25rem;
}

.emotion-7 button+.button--secondary,
.emotion-7 button+.button--primary {
  margin-left: 0.5rem;
}

section:last-child .emotion-7 {
  border-bottom: none;
}

.emotion-5 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.25;
  margin-right: 1rem;
  line-height: 1.25;
  margin-bottom: calc(0.875rem * 0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-bottom: 0;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  font-size: inherit;
  font-weight: 400;
  padding: 0;
}

.emotion-0 {
  margin-right: 0.75rem;
}

.emotion-9 {
  font-family: Motiva Sans;
  overflow: hidden;
  box-sizing: border-box;
  display: block;
  border-bottom: 1px solid #E0E1E2;
}

section:last-child .emotion-9 {
  border-bottom: none;
  border-top: 1px solid #E0E1E2;
}

<section>
  <header
    aria-controls="my-accordion-1"
    aria-expanded={true}
    className="emotion-7 emotion-8"
    onClick={[Function]}
  >
    <h1
      className="emotion-4 emotion-5 emotion-6"
      color="text.dark"
      fontSize="size4"
      fontWeight="medium"
    >
      <span
        aria-controls="my-accordion-1"
        aria-expanded={true}
        className="emotion-2 emotion-3"
      >
        <i
          className="ar ar-chevron emotion-0 emotion-1"
        />
        My Accordion
      </span>
    </h1>
  </header>
  <div
    className="emotion-9 emotion-10"
    display="block"
    id="my-accordion-1"
    overflow="hidden"
  >
    <p>
      Hello World
    </p>
  </div>
</section>
`;
